app/actions/app/server.test.ts

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createIntl} from 'react-intl';
+import {Navigation} from 'react-native-navigation';
+
+import {doPing} from '@actions/remote/general';
+import {fetchConfigAndLicense} from '@actions/remote/systems';
+import {Preferences, Screens} from '@constants';
+import DatabaseManager from '@database/manager';
+import {DEFAULT_LOCALE, getTranslations} from '@i18n';
+import SecurityManager from '@managers/security_manager';
+import WebsocketManager from '@managers/websocket_manager';
+import {getServer, getServerByIdentifier, queryAllActiveServers} from '@queries/app/servers';
+import {getConfig} from '@queries/servers/system';
+import {logError} from '@utils/log';
+import {canReceiveNotifications} from '@utils/push_proxy';
+import {alertServerAlreadyConnected, alertServerError, loginToServer} from '@utils/server';
+
+import * as Actions from './server';
+
+import type {Query} from '@nozbe/watermelondb';
+import type {ServerDatabase} from '@typings/database/database';
+import type ServersModel from '@typings/database/models/app/servers';
+
+jest.mock('@queries/app/servers');
+jest.mock('@queries/servers/system');
+jest.mock('@database/manager', () => ({
+    getServerDatabaseAndOperator: jest.fn(),
+    setActiveServerDatabase: jest.fn(),
+    getActiveServerUrl: jest.fn(),
+}));
+jest.mock('@managers/security_manager');
+jest.mock('@managers/websocket_manager');
+jest.mock('@utils/log');
+jest.mock('@utils/push_proxy');
+jest.mock('@utils/server');
+jest.mock('@actions/remote/general');
+jest.mock('@actions/remote/systems');
+jest.mock('react-native-navigation');
+
+const translations = getTranslations(DEFAULT_LOCALE);
+const intl = createIntl({locale: DEFAULT_LOCALE, messages: translations});
+const theme = Preferences.THEMES.denim;
+
+describe('initializeSecurityManager', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should return when no servers are found', async () => {
+        const mockQuery = {
+            fetch: jest.fn().mockResolvedValueOnce([]),
+        } as unknown as Query<ServersModel>;
+        jest.mocked(queryAllActiveServers).mockReturnValueOnce(mockQuery);
+        await Actions.initializeSecurityManager();
+        expect(SecurityManager.init).not.toHaveBeenCalled();
+    });
+
+    it('should initialize SecurityManager with querying configurations', async () => {
+        const servers = [{url: 'server1'}, {url: 'server2'}];
+        const mockQuery = {
+            fetch: jest.fn().mockResolvedValueOnce(servers),
+        } as unknown as Query<ServersModel>;
```

Comment:
This could be an interesting function to add to the TestHelper.

I envision something like this:

```
function mockQuery<T>(returnValue: T) {
     return {
        fetch: async () => returnValue,
        observe: of$(returnValue),
     } as unknown as Query<T>;
}
```